bgpd/bgp_attr.c

```diff
@@ -2789,6 +2780,17 @@ bgp_attr_parse_ret_t bgp_attr_parse(struct peer *peer, struct attr *attr,
 		}
 	}
 
+	/*
+	 * draft-ietf-idr-bgp-prefix-sid-27#section-3:
+	 * About Prefix-SID path attribute,
+	 * Label-Index TLV(type1) and The Originator SRGB TLV(type-3)
+	 * may only appear in a BGP Prefix-SID attribute attached to
+	 * IPv4/IPv6 Labeled Unicast prefixes ([RFC8277]).
+	 * It MUST be ignored when received for other BGP AFI/SAFI combinations.
+	 */
+	if (!attr->mp_nexthop_len || mp_update->safi != SAFI_LABELED_UNICAST)
```

Comment:
A very small optimization would be to add check for ""type == BGP_ATTR_PREFIX_SID"" to this.

Reply:
Thank you for your review!
But, I think that around this line (outside of `while() { switch (type) {} }`), bgpd already finished to parse each path-attribute's binary block. all attr's information is stored to one variable `attr` so, we can't apply this optimization.

as following.
```c
bgp_attr_parse_ret_t bgp_attr_parse(...) {
  while (BGP_INPUT_PNT(peer) < endp) {
    type = ...
    switch (type) {
    ...
    case BGP_ATTR_PREFIX_SID:
      ret = bgp_attr_prefix_sid(&attr_args);
      break;
    }
    ...
  }

  // type doesn't have valid semantics after this line.
  // each typed-attr's contect is already stored variable `attr`

  /*
   * draft-ietf-idr-bgp-prefix-sid-27#section-3:
   * About Prefix-SID path attribute,
   * Label-Index TLV(type1) and The Originator SRGB TLV(type-3)
   * may only appear in a BGP Prefix-SID attribute attached to
   * IPv4/IPv6 Labeled Unicast prefixes ([RFC8277]).
   * It MUST be ignored when received for other BGP AFI/SAFI combinations.
   */
  if (!attr->mp_nexthop_len || mp_update->safi != SAFI_LABELED_UNICAST)
    attr->label_index = BGP_INVALID_LABEL_INDEX;
}
```